services/warning/service.go

```diff
@@ -0,0 +1,87 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package warning
+
+import (
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd/pkg/deprecation"
+	"github.com/containerd/containerd/plugin"
+	"github.com/containerd/containerd/plugin/registry"
+	"github.com/containerd/containerd/plugins"
+	"github.com/containerd/containerd/services"
+)
+
+type Service interface {
+	EmitSchema1Warning()
+	EmitGoPluginLibraryWarning()
+	Warnings() []Warning
+}
+
+func init() {
+	registry.Register(&plugin.Registration{
+		Type: plugins.WarningPlugin,
+		ID:   services.WarningService,
```

Comment:
Set an actual ID here, maybe call it "deprecations" if it is specifically for deprecations. It's best not to use the "service" name or directory, as that is specifically for the "services" plugin type mirroring for serving GRPC services. A subdirectory of `plugins` would be more appropriate based on the structure we are moving to in 2.0.